### DISCUSSION TITLE
how can we integrate paytm pg on codeigniter 3 as our need

### DISCUSSION THREAD

**Discussion by jaipalsin** [codewars/codewars.com#3133] [Q&A]
I want to integrate PAYTM PAYMENT GATEWAY  into the PHP  (Codeigniter 3 mvc Framework) pattern
Paytm payment gateway supports checksum generation and verification utility which I want to use as per my requirement.
I've also completed the checksum validation utility and set up the environment by creating a Paytm library on the MVC pattern, but I can't figure out how to make the form the way I want. How to apply the Paytm PG Kit code as per my requirement

1. **My _form requirement_ is as follows:**
I want to use PG Kit code received from Paytm in my project as per my requirements. I want that when the user adds an item to the cart and goes to the checkout button, the user is redirected to the shipping data (mobile, name, address, pin code, street, etc.) submission page. Where after submitting the shipping data, as per the Paytm credentials, the user will be redirected to a callback page where he will make the payment with the total amount. Once payment is completed, the user should be shown complete order details including shipping data and information obtained from the API response.

Implement the below code as per the above 1.**form requirement** and how can it be done using 'Session/RESTful API'? Please provide me the complete code or logic for this so that I can understand it in depth in a better way

**checkout process order**

   ```
 $paramList["MID"] = PAYTM_MERCHANT_MID;
    $paramList["ORDER_ID"] = 'id'; 
    $paramList["CUST_ID"] = 'customer_id';
    $paramList["INDUSTRY_TYPE_ID"] = INDUSTRY_TYPE_ID;
    $paramList["CHANNEL_ID"] = $CHANNEL_ID;
    $paramList["TXN_AMOUNT"] = $item["subtotal"];
    $paramList["WEBSITE"] = PAYTM_MERCHANT_WEBSITE;
   
    $paramList["CALLBACK_URL"] = base_url('Checkout/payment_response');
    $paramList["MSISDN"] = 'xxxxxxxxxx'; //Mobile number of customer
    $paramList["EMAIL"] = 'xxxxxxxxxx'; //Email ID of customer
    $paramList["VERIFIED_BY"] = "MSISDN";
    $paramList["IS_USER_VERIFIED"] = "YES";
   
    $checkSum = $this->paytm->getChecksumFromArray($paramList,PAYTM_MERCHANT_KEY);
    $transaction_url = PAYTM_TXN_URL;
    $data = array('paramList' => $paramList,'transaction_url' => $transaction_url,'checkSum' => $checkSum);
   
    // ,'order_id' => $order_id
    $this->load->view('order_process',$data);
```

**callback_URL**

   ```
 $paytmChecksum = "";
    $paramList = array();
    $isValidChecksum = FALSE;
   
    $paramList = $_POST;
   
    $paytmChecksum = isset($_POST["CHECKSUMHASH"]) ? $_POST["CHECKSUMHASH"] : ""; //Sent by Paytm pg
   
    $isValidChecksum = $this->paytm->verifychecksum_e($paramList, PAYTM_MERCHANT_KEY, $paytmChecksum); //will return TRUE or FALSE string.
     
   
    if($isValidChecksum == "TRUE") {
        if ($_POST["STATUS"] == "TXN_SUCCESS") {

        //Here we can show order information

        }else{
            echo 'Transaction Failed';
        }
    }else{
        echo 'Not Valid Check Sum';
    }

```

**Comment by hobovsky** [comment]
Codewars Github is not meant to discuss general programming questions. It's meant to discuss problems and ideas related to Codewars platform.

You might have better luck asking at some general programming forum, at some PHP forum, or at forum related to PYTM or Codeigniter 3 mvc Framework itself.

I am closing this discussion as off-topic.
